pkg/database/sqlite_db.go

```diff
@@ -0,0 +1,109 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/XSAM/otelsql"
+	"github.com/golang-migrate/migrate/v4"
+	_ "github.com/golang-migrate/migrate/v4/database/sqlite3" // migration for sqlite3
+	bindata "github.com/golang-migrate/migrate/v4/source/go_bindata"
+	_ "github.com/mattn/go-sqlite3" // sqlite3 driver
+	"github.com/rs/zerolog"
+	logger "github.com/rs/zerolog/log"
+	"github.com/textileio/go-tableland/pkg/database/db"
+	"github.com/textileio/go-tableland/pkg/database/migrations"
+	"github.com/textileio/go-tableland/pkg/metrics"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// SQLiteDB represents a SQLite database.
+type SQLiteDB struct {
+	URI     string
+	DB      *sql.DB
+	Queries *db.Queries
+	Log     zerolog.Logger
+}
+
+// OpenSerializable opens a dSQLite database with only one connection open per time.
+func OpenSerializable(path string, attributes ...attribute.KeyValue) (*SQLiteDB, error) {
+	attributes = append(attributes, attribute.String("type", "serializable"))
+	return Open(path, 1, attributes...)
+}
+
+// OpenConcurrent opens a SQLite database that allows multiple connections.
+// Should be used for reads.
+func OpenConcurrent(path string, attributes ...attribute.KeyValue) (*SQLiteDB, error) {
+	attributes = append(attributes, attribute.String("type", "concurrent"))
+	return Open(path, 0, attributes...)
```

Comment:
although this is supposed to be concurrent do we still want to have an upper cap on it in order to be protective of system resources?

Reply:
Good question. Probably, but don't know what that value could be. 

Also, not sure how much a cap would help. Suppose we have a cap of 500. That means at most 500 concurrent access can happen at the `*sql.DB` layer. But at the HTTP layer, there's no limit (only the rate limiter, but that's limited by IP address) as far as I'm concerned. That means the gateway could still be receiving much more than 500 HTTP requests which would probably affect the system resources. But those HTTP request would have a bigger wait time because now the `*sql.DB` locking them. I think my point is: having a cap at the database layer does not help much if you don't have a cap at the HTTP layer, or at least a way of refusing more HTTP connections if the database is "busy".

anyway, good question. gotta think more about this